docs/tutorials/q_learning_torch.py

```diff
@@ -0,0 +1,190 @@
+        self.fc2 = nn.Linear(128, 128)
+        self.fc3 = nn.Linear(128, output_dim)
+
+    def forward(self, x):
+        x = torch.relu(self.fc1(x))
+        x = torch.relu(self.fc2(x))
+        x = self.fc3(x)
+        return x
+
+# %%%
+# In this scenario, the output dimension will indeed be two, as previously mentioned. As for the input dimension, it will be four, corresponding to the observation space of ``CartPole-v1``. 
+# Our next step is to load the dataset and set up the training loop. The ``MinariDataset`` is compatible with the PyTorch Dataset API, allowing us to load it directly using PyTorch DataLoader. 
+# However, since each episode can have a varying length, we need to pad them. 
+# To achieve this, we can utilize the `collate_fn <https://pytorch.org/docs/stable/data.html#working-with-collate-fn>`_ feature of PyTorch DataLoader. Let's create the ``collate_fn`` function:
+
+def collate_fn(batch):
+    return {
+        "id": torch.Tensor([x.id for x in batch]),
+        "seed": torch.Tensor([x.seed for x in batch]),
+        "total_timesteps": torch.Tensor([x.total_timesteps for x in batch]),
+        "observations": torch.nn.utils.rnn.pad_sequence(
+            [torch.as_tensor(x.observations) for x in batch],
+            batch_first=True
+        ),
+        "actions": torch.nn.utils.rnn.pad_sequence(
+            [torch.as_tensor(x.actions) for x in batch],
+            batch_first=True
+        ),
+        "rewards": torch.nn.utils.rnn.pad_sequence(
+            [torch.as_tensor(x.rewards) for x in batch],
+            batch_first=True
+        ),
+        "terminations": torch.nn.utils.rnn.pad_sequence(
+            [torch.as_tensor(x.terminations) for x in batch],
+            batch_first=True
+        ),
+        "truncations": torch.nn.utils.rnn.pad_sequence(
+            [torch.as_tensor(x.truncations) for x in batch],
+            batch_first=True
+        )
+    }
+
+# %%%
+# We can now proceed to load the data and create the training loop. To begin, let's initialize the DataLoader, neural network, optimizer, and loss.
+
+minari_dataset =  minari.load_dataset("CartPole-v1-random")
+dataloader = DataLoader(minari_dataset, batch_size=64, shuffle=True, collate_fn=collate_fn)
+
+env = minari_dataset.recover_environment()
+observation_space = env.observation_space
+action_space = env.action_space
+assert isinstance(observation_space, spaces.Box)
+assert isinstance(action_space, spaces.Discrete)
+
+q_net = QNetwork(np.prod(observation_space.shape), action_space.n)
+optimizer = torch.optim.Adam(q_net.parameters(), lr=0.001)
+loss_fn = nn.MSELoss()
+
+# %%%
+# We use the Mean Square Error loss, as a classic regression task. In fact, we want to predict the `Q`-values of each state-action pairs that we take from the trajectories. 
+# The training loop will look like this:
+
+num_epochs = 3
+gamma = 0.99
+
+for epoch in range(num_epochs):
+    for batch in tqdm(dataloader):
+        q_values = q_net(batch['observations'])
```

Comment:
I think it would be good to add shape annotations here, so that users can understand loss completely

Reply:
Good point, I will update